crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs

```diff
@@ -360,6 +363,15 @@ impl VisitAstPath for DependencyScanner<'_> {
     self.add_new_url(node, &*ast_path);
     node.visit_children_with_path(self, ast_path);
   }
+  fn visit_try_stmt<'ast: 'r, 'r>(
+    &mut self,
+    node: &'ast swc_core::ecma::ast::TryStmt,
+    ast_path: &mut swc_core::ecma::visit::AstNodePath<'r>,
+  ) {
+    self.in_try = true;
```

Comment:
I currently could not come up with a problem case. But the code is kind of buggy.

It's better to store the old `in_try` value and then recover it.

```rust
let old_in_try = self.in_try;
self.in_try = true
node.visit_children_with_path(self, ast_path);
self.in_try = old_in_try
```

Reply:
> I currently could not come up with a problem case. But the code is kind of buggy.
> 
> It's better to store the old `in_try` value and then recover it for the nested try-catch block.
> 
> ```rust
> let old_in_try = self.in_try;
> self.in_try = true
> node.visit_children_with_path(self, ast_path);
> self.in_try = old_in_try
> ```

Do I need to submit an mr Again?